packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -573,16 +584,43 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         }
       }
 
-      const readonly = !!this.tsParseModifier(["readonly"]);
+      const denyModifiers = [
+        "declare",
+        "abstract",
+        "private",
+        "protected",
+        "public",
+        "static",
+      ];
+      this.tsParseModifiers(
+        node,
+        [
+          "readonly",
+          "declare",
+          "abstract",
+          "private",
+          "protected",
+          "public",
+          "static",
+        ],
```

Comment:
```suggestion
        ["readonly", ...denyModifiers],
```

so that the two lists don't get out of sync.